manifests/init.pp

```diff
@@ -50,9 +50,15 @@
 
   if has_key($java::params::java, $distribution) {
     $default_package_name     = $java::params::java[$distribution]['package']
-    $default_alternative      = $java::params::java[$distribution]['alternative']
-    $default_alternative_path = $java::params::java[$distribution]['alternative_path']
-    $java_home                = $java::params::java[$distribution]['java_home']
+    if has_key($java::params::java[$distribution], 'alternative') {
+      $default_alternative = $java::params::java[$distribution]['alternative']
```

Comment:
It looks like it's not currently used for anything.  Presumably another class could use it via `$::java::java_home` to find out the value of the `JAVA_HOME` environment variable.